src/test/java/com/amazonaws/encryptionsdk/model/CiphertextHeadersTest.java

```diff
@@ -314,6 +314,24 @@ private void readUptoNonceLen(final ByteBuffer headerBuff) {
         headerBuff.get();
     }
 
+    @Test(expected = BadCiphertextException.class)
+    public void incompleteCiphertext() {
+        final Map<String, String> encryptionContext = new HashMap<String, String>(1);
+        encryptionContext.put("ENC", "CiphertextHeader Streaming Test");
+
+        final CiphertextHeaders ciphertextHeaders = createCiphertextHeaders(encryptionContext);
+        final byte[] headerBytes = ciphertextHeaders.toByteArray();
+        final ByteBuffer headerBuff = ByteBuffer.wrap(headerBytes);
+
+        readUptoVersion(headerBuff);
+
+        // set ciphertext to incomplete.
+        headerBuff.put((byte) 0);
+
+        final CiphertextHeaders reconstructedHeaders = new CiphertextHeaders();
+        reconstructedHeaders.deserialize(headerBuff.array(), 0);
```

Comment:
I *think* what is happening here is that this throws a BadCiphertextException because you mess up the buffer on L329. This results in BadCiphertextException not because the CiphertextHeaders is not complete, but because in the process of deserializing the bytes one of the values doesn't make sense (my guess would be here: https://github.com/aws/aws-encryption-sdk-java/blob/master/src/main/java/com/amazonaws/encryptionsdk/model/CiphertextHeaders.java#L205)

This means that this test is likely a duplicate of the test here: https://github.com/aws/aws-encryption-sdk-java/blob/master/src/test/java/com/amazonaws/encryptionsdk/model/CiphertextHeadersTest.java